app/controllers/admin/locations_controller.rb

```diff
@@ -0,0 +1,56 @@
+class Admin::LocationsController < Admin::BaseController
+  
+  before_action :require_location, only: [:edit, :update, :destroy]
+
+  def index
+    @locations = Location.all
+  end
+
+  def new
+    @location = Location.new
+  end
+
+  def edit
+  end
+
+  def create
+    @location = Location.new(location_params)
+    if @location.save
+      redirect_to [:edit, :admin, @location], notice: 'Created!'
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @location.update(location_params)
+      redirect_to [:edit, :admin, @location], notice: 'Updated!'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @location.destroy
```

Comment:
This will never fail, perhaps we need a before_destroy check to disallow deletion of locations that have cohorts? Worried that someone will make a critical mistake here.